borsh/src/schema/container_ext/validate.rs

```diff
@@ -53,14 +71,25 @@ fn validate_impl<'a>(
     stack.push(declaration);
     match definition {
         Definition::Primitive(_size) => {}
-        Definition::Array { elements, .. } => validate_impl(elements, schema, stack)?,
+        // arrays branch
+        Definition::Sequence {
+            length_width,
+            length_range,
+            elements,
+        } if *length_width == Definition::ARRAY_LENGTH_WIDTH
+            && length_range.clone().count() == 1 =>
```

Comment:
`length_width == 0 && length_range.len() > 1` is needed for varint and other custom encodings which have their own way of determining length.

Reply:
covered in the following arm